File: scm-csharp/scheme/primitives/PrimitiveHttpRequestP.cs
namespace scheme;

public class PrimitiveHttpRequestP : Primitive
{
    public override string Name() => "http-request?";

    public override string Info() =>
        "Syntax: (http-request? x)\n" +
        "Library: (scm net http request)\n" +
        "Description: Returns #t if x is an HTTP request object.\n" +
        "Example:\n" +
        "  (http-request? (make-http-request \"GET\" \"/\" '() #f)) => #t";

    public override object Apply(SourcePos? pos, object[] arguments)
    {
        CheckArgs(pos, arguments, 1, 1);
        return Value.IsNativeValue(arguments[0]) && Value.AsNativeValue(arguments[0]).value is SchemeHttpRequest
            ? Value.T : Value.F;
    }
}

File: scm-java/src/main/scheme/primitives/PrimitiveHttpRequestP.java
package scheme.primitives;
import scheme.*;

public class PrimitiveHttpRequestP extends Primitive {
    @Override
    public String name() { return "http-request?"; }

    @Override
    public String info() {
        return "Syntax: (http-request? x)\n" +
               "Library: (scm net http request)\n" +
               "Description: Returns #t if x is an HTTP request object.\n" +
               "Example:\n" +
               "  (http-request? (make-http-request \"GET\" \"/\" '() #f)) => #t";
    }

    @Override
    public Object apply(SourcePos pos, Object[] arguments) {
        checkArgs(pos, arguments, 1, 1);
        return Value.isNativeValue(arguments[0]) && Value.asNativeValue(arguments[0]).value instanceof SchemeHttpRequest
            ? Value.T : Value.F;
    }
}
